esphomeyaml/components/time.rst

```diff
@@ -59,6 +59,32 @@ Configuration variables:
 - **on_time** (*Optional*, :ref:`Automation <automation>`): Automation to run at specific intervals using
   a cron-like syntax. See :ref:`time-on_time`.
 
+DS1307 Time Source
+------------------
+
+With the ``DS1307`` time platform, the battery powered clock chip connected via
+:doc:`I2C </esphomeyaml/components/i2c>` bus is used to synchronize the system
+time at start-up and upon request.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    time:
+      - platform: sntp
+      - platform: ds1307
+        id: ds1307_time
+
+    interval:
+      - interval: 15min
+        then:
+          - lambda: id(ds1307_time).write_data();
```

Comment:
The user ideally shouldn't need to write an automation for a basic task like synchronizing time.

The component should call `set_interval` in setup() and register a function to write data to the sensor every 15 minutes or so (could even be a config option, where `never` would mean the time is never written, see `cv.update_interval`). In there, the call to `now()` should also be checked with `is_valid` and only then written.